test/NuGet.Clients.Tests/NuGet.Credentials.Test/SecurePluginCredentialProviderTests.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Plugins;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Credentials.Test
+{
+    public class SecurePluginCredentialProviderTests
+    {
+        [Fact]
+        public void Create_ThrowsForNullPlugin()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(
+                () => new SecurePluginCredentialProvider(null, NullLogger.Instance));
+
+            Assert.Equal("pluginDiscoveryResult", exception.ParamName);
+        }
+
+        [Fact]
+        public void Create_ThrowsForNullLogger()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(
+                () => new SecurePluginCredentialProvider(CreatePluginDiscoveryResult(), null));
+
+            Assert.Equal("logger", exception.ParamName);
+        }
+
+        [Fact]
+        public void Create_ThrowsForInvalidPlugin()
+        {
+            var exception = Assert.Throws<ArgumentException>(
```

Comment:
Validate the return value (e.g.:  check parameter name) or don't bother capturing it in the first place.